Hw03 mongodb

Alex-T-T opened the pull request:
Думаю, що все Ок! 
Але якщо ні, то не біда, виправимо на Ок! ))))

Alex-T-T committed:
add MongoDB

Alex-T-T committed:
.env, dotenv.config

1) replacec contsctSchema from src/models/contact.js to src/models/contactSchema
2) add .env.example
3) replace require('dotenv').config() from server.js to app.js
4) fix mongoose.connection

Alex-T-T committed:
add controllerCheck

add folder "Utils"
add utils/controllerCheck
use controllerCheck in src/routes/api/contacts for checking contactsCotroller in router

Alex-T-T committed:
update app.use error

Alex-T-T committed:
update validation

1) update validation Joi
2) contactSchema.js - add to contactSchema option:  writeConcern

Alex-T-T committed:
check for valid ID

1) add isValidId.js
2) add handleValidation.js
3) add RequestErrors.js

Alex-T-T committed:
Update contacts.js

1) add isValidId ti router.patch

Alex-T-T committed:
throw RequestError

1) add RequestError to contactsControllers.js
2) fix check for field "favorite" in const patchContactController (line 49)

Alex-T-T committed:
Update server.js

change connection DB & server start

OnyshchenkoO commented on src/controllers/contactsController.js:
```diff
@@ -3,54 +3,65 @@ const {listContacts,
     res.status(200).json({updatedContact, message: 'successfully updated'});
 };
 
+const patchContactController = async (req, res, next) => {
```
Це не дуже правильна назва контролера, бо ти оновлюєш лише 1 поле контакту, тому я б назва updateContactFavoriteController

Ще у валідаційніх схемі теж би змінив назву

Alex-T-T replied:
Перейменував!

OnyshchenkoO commented on server.js:
```diff
@@ -1,8 +1,17 @@
+
+mongoose.connect(DB_HOST)
+  .then(
+    console.log("Database connection successful"),
+    app.listen(PORT, () => console.log(`Server running. Use our API on port: ${PORT}`)),
```
А чого ти передаєш, як 2 різних аргументи колбеку?

Alex-T-T replied:
змінив

OnyshchenkoO commented on src/middlewares/validationMiddlware.js:
```diff
@@ -39,14 +39,34 @@ module.exports = {
+    if (validationResult.error) {
+        return res.status(400)
+            .json({
+            'message': validationResult.error.details[0].message,
```
validationResult.error.details[0].message - воно покаже помилку лише для першої помилки чи для всіх? Якщо лише для першої, то я б краще змінив, щоб показувати всі помилки разом. Але як бажаєш

Alex-T-T replied:
я змінив на "validationResult.error.message", але в кінцевому результаті і такий, і "validationResult.error.details[0].message" такий варіант виводять однаковий результат.
У такому варіанті validationResult.error.details[0].message - validationResult.error.details - повертає масив з одним обʼєктом (в якому поле message дублюється з validationResult.error.message )

Alex-T-T committed:
after mentor`s review

1) change then() in server.js
2) rename function "patchContactController" to "updateContactFavoriteController"
3) update validation error

Alex-T-T committed:
fix validationSchema

rename "patchContactValidation" to "updateContactFavoriteValidation"

Alex-T-T committed:
Update validationMiddlware.js
